examples/demo-orderapi.oas2.yaml

```diff
@@ -0,0 +1,384 @@
+      - $ref: '#/definitions/halResource'
+      - $ref: '#/definitions/order'
+    example:
+      _links:
+        self:
+          href: /orders/1234
+
+        edit:
+          href: /orders/1234
+
+        delete:
+          href: /orders/1234
+
+        profile:
+          href: https://adidas-group.com/gdm/OMS
+
+      orderNumber: 1234
+      itemCount: 42
+      status: pending
+
+  # Partial Order
+  orderPatch:
+    type: object
+    allOf:
+      - $ref: '#/definitions/order'
+    example:
+      status: cancelled
+      orderNumber: 1
+      itemCount: 2
+
+  # Order
+  order:
+    type: object
+    properties:
+      orderNumber:
+        type: number
+      itemCount:
+        type: number
+      status:
+        type: string
+    required:
+      - orderNumber
+      - itemCount
+    example:
+      orderNumber: 42
+      itemCount: 3
+      status: pending
+
+  # Order Not Found Error
+  orderNotFound:
+    type: object
+    allOf:
+      - $ref: '#/definitions/problemDetail'
+    example:
+      title: 'Not Found'
+      detail: 'Cannot find the requested order'
+      status: 404
+      instance: '/orders/1234'
+
+  #
+  # -- DO NOT EDIT BEYOND THIS POINT --
+  #
+
+  # Media Types Definitions
+
+  #
+  # application/problem+json
+  #
+  problemDetail:
+    type: object
+    properties:
+      type:
+        type: string
+      title:
+        type: string
+      status:
+        type: number
+      detail:
+        type: string
+      instance:
+        type: string
+    required:
+      - title
+      - detail
+
+  #
+  # application/hal+json defintion
+  #
+  halLinkObject:
+    type: object
+    required:
+      - href
+    properties:
+      href:
+        type: string
+      templated:
+        type: boolean
+      #
+      # WARN: Some version of OAS parser can have problem with property being called "type"
+      # resulting in the error 'Schema "type" key must be a string'.
+      #
+      # For this reason we have commented out the "type" property of the link object.
+      # See https://tools.ietf.org/html/draft-kelly-json-hal-08#section-5.3 for the definiton of
+      # HAL's link object type property.
+      #
+      # type:
+      #  type: string
+      deprecation:
+        type: string
+      name:
+        type: string
+      profile:
+        type: string
+      title:
+        type: string
+      hreflang:
+        type: string
+
+  halCuriesLink:
+    title: HAL Curies Link
+    allOf:
+      - type: object
+        properties:
+          templated:
+            enum:
+              - true
+        required:
+          - templated
+      - $ref: '#/definitions/halLinkObject'
+
+  halResource:
+    title: HAL Resource Object
+    type: object
+    properties:
+      _links:
+        type: object
+        additionalProperties:
+          # WARN: Should be "anyOf" but "anyOf" isn't supported in Swagger 2.0
+          allOf:
+            - $ref: '#/definitions/halLinkObject'
+            - type: array
```

Comment:
This is what causes these 3 `valid-example-in-definitions` errors.
Basically, what we are trying to assert here is that a given value is both an object (as specified in '#/definitions/halLinkObject') and an array.
I guess `oneOf` would the job here?

Reply:
after a discussion with @zdne - it might not make sense to check validity of examples in case the schemas use schema elements not supported by OAS2. In this case just presence of examples may be the most we can do. We have the schema defined correctly (with `anyOf` here: https://supermodel.io/adidas/api/HAL?view=schema) and then converted to be oas2 compatible, using `allOf`, which may or may not be correct. There is no way, as far as I know, to link those two.
Any comments @P0lip @philsturgeon?